extension-base/src/main/java/com/azure/autorest/extension/base/plugin/JavaSettings.java

```diff
@@ -639,6 +639,27 @@ public boolean shouldUseIterable() {
         return serviceVersions;
     }
 
+    private Map<String, String> llcProperties;
+
+    public Map<String, String> getLlcProperties() {
+        if (llcProperties == null) {
+            llcProperties = new HashMap<>();
+        }
+        llcProperties.put("artifact-id", getArtifactId());
+        llcProperties.put("service-name", getServiceName());
+        llcProperties.put("namespace-path", getPackage().replaceAll("\\.", "/"));
+        if (!llcProperties.containsKey("url-fragment") && getServiceName() != null) {
+            llcProperties.put("url-fragment", getServiceName().replaceAll("\\s", "-").toLowerCase() + "-java-samples");
+        }
+        return llcProperties;
+    }
+
+    private final boolean llcGenerateNonCode;
+
+    public boolean isLlcGenerateNonCode() {
```

Comment:
Why is this its own setting, instead of being merged into `llc-properties`?

Reply:
This parameter controls whether generating non java file while `llc-properties` stores data that should be filled to resource files.